File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.ts
function strStr(haystack: string, needle: string): number {
    if (needle.length === 0) return 0; // Edge case: empty needle
    if (needle.length > haystack.length) return -1; // Needle longer than haystack

    for (let i = 0; i <= haystack.length - needle.length; i++) {
        let match = true;
        for (let j = 0; j < needle.length; j++) {
            if (haystack[i + j] !== needle[j]) {
                match = false;
                break;
            }
        }
        if (match) {
            return i; // Return the starting index of the match
        }
    }

    return -1; // No match found
}

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) return 0;
        if (haystack.length() < needle.length()) return -1;

        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
            int j;
            int curI = i;

            for (j = 0; j < needle.length(); j++) {
                if (haystack.charAt(curI) != needle.charAt(j)) break;
                curI++;
            }

            if (j == needle.length()) return i;
        }
        return -1;
    }
}
